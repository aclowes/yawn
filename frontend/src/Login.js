import React from 'react';
import {
  FormGroup, FormControl, ControlLabel, Button, Alert, Panel, Grid, Row, Col
} from 'react-bootstrap';

import {YawnNavBar} from './utilities';
import API from './API'

export default class Login extends React.Component {
  constructor(props) {
    super(props);
    this.state = {username: '', password: '', error: null};
  }

  componentDidMount() {
    document.title = `YAWN - Login`;
  }

  handleChange = (event) => {
    this.setState({[event.target.id]: event.target.value});
  };

  handleSubmit = (event) => {
    event.preventDefault();
    const credentials = {
      username: this.state.username,
      password: this.state.password
    };
    API.patch('/api/users/login/', credentials, (payload, error) => {
      if (error) {
        this.setState({error});
      } else {
        this.props.router.push('/');
      }
    });
  };

  render() {
    return (
      <YawnNavBar>
        <Grid>
          <Row className="show-grid">
            <Col xs={1} md={4}/>
            <Col xs={4} md={4}>
              <Panel header="Login">
                <form onSubmit={this.handleSubmit}>
                  {this.state.error && <Alert bsStyle="danger">{this.state.error}</Alert>}
                  <FormGroup controlId="username">
                    <ControlLabel>Username</ControlLabel>
                    <FormControl type="text" value={this.state.username}
                                 onChange={this.handleChange}
                    />
                  </FormGroup>
                  <FormGroup controlId="password">
                    <ControlLabel>Password</ControlLabel>
                    <FormControl type="password" value={this.state.password}
                                 onChange={this.handleChange}
                    />
                  </FormGroup>
                  <Button type="submit" bsStyle="primary">Login</Button>
                </form>
              </Panel>

            </Col>
            <Col xs={1} md={4}/>
          </Row>
        </Grid>
      </YawnNavBar>
    );
  }
}
