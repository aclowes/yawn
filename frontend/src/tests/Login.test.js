import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import Login from '../Login';

it('Login renders', () => {
  const component = ReactTestRenderer.create(<Login />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
