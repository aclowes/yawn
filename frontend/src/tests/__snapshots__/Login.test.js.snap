// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders 1`] = `
<div>
  <nav
    className="navbar navbar-default"
  >
    <div
      className="container"
    >
      <div
        className="navbar-header"
      >
        <span
          className="navbar-brand"
        >
          YAWN
        </span>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div
      className="container"
    >
      <div
        className="show-grid row"
      >
        <div
          className="col-md-4 col-xs-1"
        />
        <div
          className="col-md-4 col-xs-4"
        >
          <div
            className="panel panel-default"
            id={undefined}
          >
            <div
              className="panel-heading"
            >
              Login
            </div>
            <div
              className="panel-body"
            >
              <form
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                    htmlFor="username"
                  >
                    Username
                  </label>
                  <input
                    className="form-control"
                    id="username"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                    htmlFor="password"
                  >
                    Password
                  </label>
                  <input
                    className="form-control"
                    id="password"
                    onChange={[Function]}
                    type="password"
                    value=""
                  />
                </div>
                <button
                  className="btn btn-primary"
                  disabled={false}
                  type="submit"
                >
                  Login
                </button>
              </form>
            </div>
          </div>
        </div>
        <div
          className="col-md-4 col-xs-1"
        />
      </div>
    </div>
  </div>
</div>
`;
